### PR TITLE
#44: o.l.c.m.HasValuesMatching

### PR DESCRIPTION
**Pull request by dgroup**
#44

### CODE REVIEW

**Comment by dgroup** [comment]
@llorllale, could you please also assign the #44 to me also?
Looks like the 0crat didn't assign the tasks to me because I have no enough points of reputation yet.

**Comment by 0crat** [comment]
Job [#45](https://github.com/llorllale/cactoos-matchers/issues/45) is now in scope, role is `REV`

File: src/main/java/org/llorllale/cactoos/matchers/HasValuesMatching.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+ * @since 1.0.0
+ * @checkstyle ProtectedMethodInFinalClassCheck (200 lines)
+ */
+public final class HasValuesMatching<X> extends
+    TypeSafeDiagnosingMatcher<Iterable<X>> {
+
+    /**
+     * The description of testing scenario.
+     */
+    private final String description;
```
Comment (llorllale) [inline src/main/java/org/llorllale/cactoos/matchers/HasValuesMatching.java]: @dgroup I don't think users should be providing this value

Comment (dgroup) [reply src/main/java/org/llorllale/cactoos/matchers/HasValuesMatching.java]: @llorllale, done.

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job `gh:llorllale/cactoos-matchers#45` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#45](https://github.com/llorllale/cactoos-matchers/issues/45) is now out of scope

**Pull request merged**